src/power_button.cpp

```diff
@@ -0,0 +1,40 @@
+#include <avr/io.h>
+#include <util/delay.h>
+#include "power_button.h"
+#include "bit.h"
+
+#define POWER_BUTTON_PIN PINB3
+#define DEBOUNCE_DELAY 100 // CHANGE THIS
+
+// This function should initialize the power button. This should mostly involve initializing the pins and other register values.
+void setup_power_button(void) {
+    // Set the push button as output and initialise it to logic high
+    BIT_SET(DDRB, POWER_BUTTON_PIN);
+    BIT_SET(PORTB, POWER_BUTTON_PIN);
+
+    // Set the push button as input
+    BIT_CLEAR(DDRB, POWER_BUTTON_PIN);
+}
+
+// This function should return true if the power button is pressed and false otherwise.
+// This function should be debounced. If the button status isn't changed from the last known value it should return immediately.
+// Otherwise, it should wait so the output can be debounced (verified) and then return the new button status.
+bool is_pressed_power_button(void) {
```

Comment:
The comment indicate this should be debounce, but it doesn't appear to be. Can you resolve this.

Reply:
Right. On it.